python/MooseDocs/test/__init__.py

```diff
@@ -53,6 +57,22 @@ def __setup(self, reader=None, renderer=None, extensions=None, executioner=None)
         self.__translator.init()
         self.__translator.execute() # This is required to setup the meta data
 
+    def setUp(self):
+        # Wrap mooseutils.runExe so that we don't call the application
+        # with the same arguments more than once. The first time with
+        # a given set of arguments, it'll run the app. Every time after
+        # that, it'll use the cache
+        def run_exe_cached(app_path, args):
+            key = app_path + ''.join(args)
+            cache = MooseDocsTestCase.RUN_EXE_CACHE
+            if key not in cache:
+                result = MooseDocsTestCase.ORIG_RUN_EXE(app_path, args)
+                cache[key] = result
+            return cache[key]
+        patcher = patch.object(mooseutils, 'runExe', wraps=run_exe_cached)
```

Comment:
This is some next level stuff

Reply:
Python mock is both wonderful and horrifying